core/computed/main-resource.js

```diff
@@ -22,7 +22,16 @@ class MainResource {
     const {mainDocumentUrl} = data.URL;
     if (!mainDocumentUrl) throw new Error('mainDocumentUrl must exist to get the main resource');
     const requests = await NetworkRecords.request(data.devtoolsLog, context);
-    const mainResource = NetworkAnalyzer.findResourceForUrl(requests, mainDocumentUrl);
+
+    const mainResourceRequests = requests.filter(request =>
+      UrlUtils.equalWithExcludedFragments(request.url, mainDocumentUrl)
+    );
+    // We could have more than one record matching the main doucment url,
+    // if the page did `location.reload()`. Since `mainDocumentUrl` refers to the _last_
+    // document request, we should return the last candidate here. Besides, the browser
+    // would have evicted the first request by the time `MainDocumentRequest` (a consumer
+    // of this computed artifact) attempts to fetch the contents, resulting in a protocol error.
+    const mainResource = mainResourceRequests[mainResourceRequests.length - 1];
```

Comment:
There are a couple other places where we get the main document request this way:
https://github.com/GoogleChrome/lighthouse/blob/1b843de669e18c144422561010985d0dde05d2c6/core/lib/navigation-error.js#L119
https://github.com/GoogleChrome/lighthouse/blob/1b843de669e18c144422561010985d0dde05d2c6/core/computed/page-dependency-graph.js#L409-L414
Should we be getting the last request in these places as well?